ipfx/sweep_props.py

```diff
@@ -0,0 +1,51 @@
+import logging
+
+
+def assign_sweep_states(manual_sweep_states, auto_sweep_states, sweep_features):
+
+    # start with auto state assignment
+    sweep_states = { s["sweep_number"]:s["passed"] for s in auto_sweep_states }
+
+    # override when manual values are available
+    for mss in manual_sweep_states:
+        sn = mss["sweep_number"]
+        logging.debug("Overriding sweep state for sweep %d from %s to %s", sn, str(sweep_states[sn]), mss["passed"])
+        sweep_states[sn] = mss["passed"]
+
+    # assign sweep state to all sweeps
+    for sweep in sweep_features:
+        sn = sweep["sweep_number"]
+        if sn in sweep_states:
+            sweep["passed"] = sweep_states[sn]
+        else:
+            logging.warning("could not find QC state for sweep number %d", sn)
+
+
+def drop_incomplete_sweeps(sweep_features):
+
+    sweep_features[:] = [s for s in sweep_features if s["completed"]]
```

Comment:
In general, it's better to have functions that have a single purpose
The assign_sweep_states is doing two things. Assigning sweep_states andd sweep["passed]

Reply:
Done. thnx.